aten/src/ATen/native/WeightNorm.cpp

```diff
@@ -57,7 +57,9 @@ std::tuple<Tensor,Tensor> weight_norm_cpu(
   const auto dtype = g.scalar_type() == at::ScalarType::BFloat16 ?
       at::ScalarType::Float : g.scalar_type();
   auto norm = at::empty_strided(g.sizes(), g.strides(), g.options().dtype(dtype));
-  weight_norm_stub(kCPU, w, norm, v, g, dim);
+  if (v.numel() && g.numel()) {
```

Comment:
What is the case where one would be empty but not the other? Wouldn't that leave one of the outputs uninitialized?

Reply:
Hmm, may be it needs to be put down the stack. 
Also, this op currently does not adhere to the expected norm rule, where even an empty tensor results in a non-empty norm, i.e. `torch.rand(3, 0).norm(1).shape` is `torch.Size([])`